src/pyproject_fmt/formatter/pep508.py

```diff
@@ -10,36 +12,19 @@
 REQ_REGEX = re.compile(r"(===|==|!=|~=|>=?|<=?|@)\s*([^,]+)")
 
 
-def _req_base(lib: str) -> str:
-    match = re.match(BASE_NAME_REGEX, lib)
-    if match is None:
-        raise ValueError(repr(lib))
-    return match.group(0)
-
-
-def _normalize_lib(lib: str) -> str:
-    base = _req_base(lib)
-    values = sorted(
-        (f"{m.group(1)}{m.group(2)}" for m in REQ_REGEX.finditer(lib)), key=lambda c: ("<" in c, ">" in "c", c)
-    )
+def normalize_req(req: str) -> str:
+    parsed = Requirement(req)
+    orig_specifiers = str(parsed.specifier)
+    values = [f"{m.group(1)}{m.group(2)}" for m in REQ_REGEX.finditer(orig_specifiers)]
     if values:  # strip .0 version
         while values[0].endswith(".0") and (values[0].startswith(">=") or values[0].startswith("==")):
             values[0] = values[0][:-2]
-    return f"{base}{','.join(values)}"
-
-
-def normalize_req(req: str) -> str:
-    lib, _, envs = req.partition(";")
-    normalized = _normalize_lib(lib)
-    envs = envs.strip()
-    if not envs:
-        return normalized
-    return f"{normalized};{envs}"
+    return str(parsed).replace(orig_specifiers, ",".join(values))
```

Comment:
Once I finished doing the changes in `_normalize_lib` to correctly process the more complex examples from the PEP 508, I noticed that the only thing `normalize_req` was doing was to split the environment markers and call `_normalize_lib`.

Since the changed `_normalize_lib` (that uses `packaging.requirements.Requirement`) is now able to handle strings even if they contain the environment markers part. The only code left here was:

```python
def normalize_req(req: str) -> str:
    return _normalize_lib(req)
```

At that point in time I decided to collapse the two functions in only one...

Please let me know if you would like to have this implemented in a different way.